src/planning.ts

```diff
@@ -62,284 +63,334 @@ export const defaultPlanOptions: PlanOptions = {
+  penLiftDuration: number
+  penDropDuration: number
 }
 
-export const Device = {
-  Axidraw: {
-    stepsPerMm: 5,
+export const Device = (hardware = 'v3'): Device => {
+  if (hardware === 'brushless') return AxidrawBrushless
+  return Axidraw
+}
 
-    // Practical min/max that you might ever want the pen servo to go on the AxiDraw (v2)
-    // Units: 83ns resolution pwm output.
-    // Defaults: penup at 12000 (1ms), pendown at 16000 (1.33ms).
-    penServoMin: 7500,  // pen down
-    penServoMax: 28000, // pen up
+export interface Device {
+  stepsPerMm: number
+  // Practical min/max that you might ever want the pen servo to go on the AxiDraw
+  // Units: 83ns resolution pwm output.
+  penServoMin: number // pen down
+  penServoMax: number // pen up
+  penPctToPos: (pct: number) => number
+}
 
-    penPctToPos(pct: number): number {
-      const t = pct / 100.0;
-      return Math.round(this.penServoMin * t + this.penServoMax * (1 - t));
-    }
+// Defaults: penup at 12000 (1ms), pendown at 16000 (1.33ms).
+const Axidraw: Device = {
+  stepsPerMm: 5,
+
+  penServoMin: 7500, // pen down
+  penServoMax: 28000, // pen up
+
+  penPctToPos (pct: number): number {
+    const t = pct / 100.0
+    return Math.round(this.penServoMin * t + this.penServoMax * (1 - t))
   }
-};
+}
+
+// brushless servo (https://shop.evilmadscientist.com/productsmenu/else?id=56)
+const AxidrawBrushless: Device = {
+  stepsPerMm: 5,
+
+  penServoMin: 5400, // pen down
+  penServoMax: 12600, // pen up
+
+  penPctToPos (pct: number): number {
+    const t = pct / 100.0
+    return Math.round(this.penServoMin * t + this.penServoMax * (1 - t))
+  }
+}
 
 export const AxidrawFast: ToolingProfile = {
   penDownProfile: {
-    acceleration: 200 * Device.Axidraw.stepsPerMm,
-    maximumVelocity: 50 * Device.Axidraw.stepsPerMm,
-    corneringFactor: 0.127 * Device.Axidraw.stepsPerMm
+    acceleration: 200 * Axidraw.stepsPerMm,
+    maximumVelocity: 50 * Axidraw.stepsPerMm,
+    corneringFactor: 0.127 * Axidraw.stepsPerMm
   },
   penUpProfile: {
-    acceleration: 400 * Device.Axidraw.stepsPerMm,
-    maximumVelocity: 200 * Device.Axidraw.stepsPerMm,
+    acceleration: 400 * Axidraw.stepsPerMm,
+    maximumVelocity: 200 * Axidraw.stepsPerMm,
     corneringFactor: 0
   },
-  penUpPos: Device.Axidraw.penPctToPos(50),
-  penDownPos: Device.Axidraw.penPctToPos(60),
+  penUpPos: Axidraw.penPctToPos(50),
+  penDownPos: Axidraw.penPctToPos(60),
   penDropDuration: 0.12,
-  penLiftDuration: 0.12,
-};
+  penLiftDuration: 0.12
+}
+
+export const AxidrawBrushlessFast: ToolingProfile = {
+  penDownProfile: {
+    acceleration: 200 * Axidraw.stepsPerMm,
+    maximumVelocity: 50 * Axidraw.stepsPerMm,
+    corneringFactor: 0.127 * Axidraw.stepsPerMm
+  },
+  penUpProfile: {
+    acceleration: 400 * Axidraw.stepsPerMm,
+    maximumVelocity: 200 * Axidraw.stepsPerMm,
+    corneringFactor: 0
+  },
+  penUpPos: Axidraw.penPctToPos(50),
+  penDownPos: Axidraw.penPctToPos(60),
```

Comment:
Should all the `Axidraw` here be `AxidrawBrushless`?

Reply:
oops